libsharedringbuffer/include/shared_ringbuffer/shared_ringbuffer.h

```diff
@@ -0,0 +1,240 @@
+
+    return 0;
+}
+
+/**
+ * Enqueue an element into an available ring buffer.
+ * This indicates the buffer address parameter is currently available for use. 
+ *
+ * @param ring Ring handle to enqueue into.
+ * @param buffer address into shared memory where data is stored.
+ * @param len length of data inside the buffer above.
+ * @param cookie optional pointer to data required on dequeueing.
+ *
+ * @return -1 when ring is empty, 0 on success.
+ */
+static inline int enqueue_avail(ring_handle_t *ring, uintptr_t addr, size_t len, void *cookie) 
+{
+    return enqueue(ring->avail_ring, addr, len, cookie);
+}
+
+/**
+ * Enqueue an element into a used ring buffer.
+ * This indicates the buffer address parameter is currently in use.
+ *
+ * @param ring Ring handle to enqueue into.
+ * @param buffer address into shared memory where data is stored.
+ * @param len length of data inside the buffer above.
+ * @param cookie optional pointer to data required on dequeueing.
+ *
+ * @return -1 when ring is empty, 0 on success.
+ */
+static inline int enqueue_used(ring_handle_t *ring, uintptr_t addr, size_t len, void *cookie) 
+{
+    return enqueue(ring->used_ring, addr, len, cookie);
+}
+
+/**
+ * Dequeue an element from an available ring buffer.
+ *
+ * @param ring Ring handle to dequeue from.
+ * @param buffer pointer to the address of where to store buffer address.
+ * @param len pointer to variable to store length of data dequeueing.
+ * @param cookie pointer optional pointer to data required on dequeueing.
+ *
+ * @return -1 when ring is empty, 0 on success.
+ */
+static inline int dequeue_avail(ring_handle_t *ring, uintptr_t *addr, size_t *len, void **cookie) 
+{
+    return dequeue(ring->avail_ring, addr, len, cookie);
+}
+
+/**
+ * Dequeue an element from a used ring buffer.
+ *
+ * @param ring Ring handle to dequeue from.
+ * @param buffer pointer to the address of where to store buffer address.
+ * @param len pointer to variable to store length of data dequeueing.
+ * @param cookie pointer optional pointer to data required on dequeueing.
+ *
+ * @return -1 when ring is empty, 0 on success.
+ */
+static inline int dequeue_used(ring_handle_t *ring, uintptr_t *addr, size_t *len, void **cookie) 
+{
+    return dequeue(ring->used_ring, addr, len, cookie);
+}
+
+/**
+ * Dequeue an element from a ring buffer.
+ * This function is intended for use by the driver, to collect a pointer
+ * into this structure to be passed around as a cookie. 
+ *
+ * @param ring Ring buffer to dequeue from.
+ * @param addr pointer to the address of where to store buffer address.
+ * @param len pointer to variable to store length of data dequeueing.
+ * @param pointer pointer to store the address to this particular entry.
+ *
+ * @return -1 when ring is empty, 0 on success.
+ */
+static int driver_dequeue(ring_buffer_t *ring, uintptr_t *addr, size_t *len, void **pointer)
+{
+    if (!((ring->write_idx - ring->read_idx) % CONFIG_LIB_SHARED_RINGBUFFER_DESC_COUNT)) {
```

Comment:
We could simply use our helper function:
```suggestion
    if (ring_empty(ring) {
```